agent-inject/agent/annotations.go

```diff
@@ -550,6 +558,22 @@ func (a *Agent) agentCacheEnable() (bool, error) {
 	return strconv.ParseBool(raw)
 }
 
+func (a *Agent) agentCachePersist() bool {
+	if a.VaultAgentCache.Enable && a.PrePopulate && !a.PrePopulateOnly {
```

Comment:
Determining whether we need persist like so may be fine for now, though I do wonder if we'd have to expose this through annotations down the road.